### ISSUE TITLE
Bug: [no-unnecessary-condition] should not say that falsey bigint is truthy

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10197]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.3&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3gMyhNIAngEICWA5gJIoYBc8ADCgNwBQp28AFLvkWVRoBKeAG9WAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRQA7iKaRwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const falseyBigInt: 0n;
if (falseyBigInt) {
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Unnecessary conditional, value is always falsy.

### Actual Result

Unnecessary conditional, value is always truthy.

### Additional Info

This is a bug with this line. 

https://github.com/typescript-eslint/typescript-eslint/blob/373c63a40ccb6c5306999a0bd4cdb9cd5348988f/packages/eslint-plugin/src/rules/no-unnecessary-condition.ts#L32

The `.value` field returns a `PseudoBigInt` _object_ that looks like `{ negative?: boolean, base10Value: string }`, rather than a `bigint` itself (for legacy TS support reasons). So checking for truthiness is incorrect; we need to convert to a real bigint.

**Comment by kirkwaiblinger** [comment]
Funny enough, TS itself has the same bug, but for completely different root cause: https://github.com/microsoft/TypeScript/issues/60320.

\*sigh\* guess no one gives bigints much love 😆
